### PR TITLE
notifications, prefilled schedule, reset password, prefilled form

### PR DESCRIPTION
**Pull request by leoenix**

### CODE REVIEW

File: client/src/__tests__/API_methods.test.js [hunk 1]
```diff
@@ -331,13 +331,6 @@ describe('testing /retrieveAllNotifications', () => {
   })
 });
 
-//Verifies get the total number of notifications command
```
Comment (JeffreyCHChan) [inline client/src/__tests__/API_methods.test.js:334]: is there a reason we removed this test?

**Review by JeffreyCHChan — COMMENTED** [review]
some minor things that need to be addressed, no refactoring needed just clarifications on comments and database changes

File: client/src/__tests__/API_methods.test.js (hunk 1 above)
Comment (leoenix) [reply client/src/__tests__/API_methods.test.js:334]: @jongjonathan you know about this?

File: server/Controllers/DoctorController.js [hunk 2]
```diff
@@ -190,6 +190,28 @@ DoctorController.get("/statusCountAllPatients", (req, res) => {
         }
     })
 });
+DoctorController.get("/statusCountMyPatients", (req, res) => {
+    let doctorID = req.query["id"];
+    //parameters:
```
[the remark this answers is not in the document]
Comment (leoenix) [reply server/Controllers/DoctorController.js:195]: yes boss

File: server/Controllers/NotificationController.js [hunk 3]
```diff
@@ -42,6 +42,23 @@ NotificationController.get("/retrieveAllNotifications", (req, res) => {
...
 });
 
+NotificationController.get("/retrieveFormNotifications", (req, res) => {
+    let doctorID = req.query["id"];
+    console.log(doctorID);
+    //parameters: DoctorID
+//returns: FName, LName, aptDate, StartTime,EndTime
+    let state = "SELECT Upatient.Fname, Upatient.Lname, Upatient.ID, Hi.InfoTimestamp " +
+        "FROM 390db.healthinformation Hi, 390db.users Upatient, 390db.doctors D, 390db.patients P " +
+        "Where Hi.PatientID = Upatient.ID AND D.ID = ? AND P.id=Upatient.id AND P.doctorID = D.id AND Hi.Notification = 1;"
```
[the remark this answers is not in the document]
Comment (leoenix) [reply server/Controllers/NotificationController.js:52]: new db 30-03-2022 9:09pm

**Review by JeffreyCHChan — APPROVED** [review]

File: client/src/__tests__/API_methods.test.js (hunk 1 above)
Comment (malhimaira) [reply client/src/__tests__/API_methods.test.js:334]: okay make sure this isn't removed @alexbara2000 , don't accept change fro this part?

File: client/src/pages/DoctorDashboard.js [hunk 4]
```diff
@@ -1,327 +1,452 @@
...
+    const handleApptMask = (index) => (event) => {
+        event.preventDefault();
+        Axios.put("http://localhost:8080/maskApptNotification", {
+            ID: notificationsList[index].ID,
+        }).then(() => {
+            window.location.reload()
+        }).catch((err) => console.log(err));
+    }
+
+    const handleFormMask = (index) => (event) => {
```
Comment (malhimaira) [inline client/src/pages/DoctorDashboard.js:48]: add comment!

File: client/src/pages/ForgetPassword.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import LockOpenTwoToneIcon from '@mui/icons-material/LockOpenTwoTone';
+import { Container, Typography, Box, TextField, CssBaseline, Button, Avatar } from '@mui/material';
+import { Component } from "react";
+import validator from 'validator';
+import Axios from "axios";
+
+
```
Comment (malhimaira) [inline client/src/pages/ForgetPassword.js:8]: this looks greta, just keep spacing consistent!

File: client/src/pages/PreviousSymptoms.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import {
+    Container,
```
Comment (malhimaira) [inline client/src/pages/PreviousSymptoms.js:3]: fix format to one line pls!

Comment (malhimaira) [reply client/src/pages/PreviousSymptoms.js:3]: also is this page outdated with the one mike updated?

**Review by malhimaira — APPROVED** [review]
Great work! I made some suggestions, mostly with regards to spacing and comments. This can be added as part of refactoring after merging 👍 also pls confirm my comment about previous symptoms  or am I mixing up symptom form pages with this?

File: client/src/pages/PreviousSymptoms.js (hunk 6 above)
Comment (malhimaira) [reply client/src/pages/PreviousSymptoms.js:3]: nvm im silly this is the one for doctor view

File: client/src/__tests__/API_methods.test.js (hunk 1 above)
Comment (leoenix) [reply client/src/__tests__/API_methods.test.js]: i added the test back

File: client/src/pages/DoctorDashboard.js (hunk 4 above)
Comment (leoenix) [reply client/src/pages/DoctorDashboard.js:49]: comments added!

File: client/src/pages/DoctorDashboard.js [hunk 7]
```diff
@@ -1,327 +1,452 @@
...
+                                        <CardHeader
+                                            avatar={
+                                                <Avatar aria-label="">
+                                                    {key + 1}
+                                                </Avatar>
+                                            }
+                                            //display patient name and appointment time
+                                            title={"Patient Name: " + val.Fname + " " + val.Lname}
+                                            subheader={"Appointment Time: " + val.aptDate + " " + val.startTime + " to " + val.endTime}
+                                        /> <IconButton aria-label="clear" onClick={handleApptMask(key)}>
```
Comment (alexbara2000) [inline client/src/pages/DoctorDashboard.js]: I don't think you can press x on a notification if it is an appointment. could you test it out and let me know if it was planned?

**Review by alexbara2000 — APPROVED** [review]
Everything looks amazing and works properly. Please read my 2 comments and answer them before merging

File: client/src/pages/DoctorDashboard.js (hunk 7 above)
Comment (leoenix) [reply client/src/pages/DoctorDashboard.js]: good find! i fixed it. although we might not want to allow this functionality? whatever you/the others think

File: client/src/pages/ForgetPassword.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        this.state = { msg: '', email: '' }
+
+        // Binding keywords keyword
+        this.handleClick = this.handleClick.bind(this)
+    }
+
+    handleClick() {
+        //Validating the email
+        if (validator.isEmail(this.state.email)) {
+            this.setState({ msg: 'Your recovery password has been sent if your email is found in our database!' })
```
[the remark this answers is not in the document]
Comment (leoenix) [reply client/src/pages/ForgetPassword.js:24]: what about "Your recovery password has been sent if your email is found in our system" ? or "found in our application"?

idea was to show that we have thought of the fact that their email is not existing. and if a user doesn't receive an email, then from that statement they can understand that they might not be using the right email 

i don't really mind too much either way so lmk what you think and i'll make the changes

[the remark this answers is not in the document]
Comment (alexbara2000) [reply client/src/pages/ForgetPassword.js:24]: I will let Jon decide how he wants to say it, but for the demo we can easily show him the code if he doesn't believe us.

File: client/src/pages/DoctorDashboard.js (hunk 7 above)
Comment (alexbara2000) [reply client/src/pages/DoctorDashboard.js]: Up to you how you want to implement it, might be a good idea to only remove this notification after the appointment passed. Is that feasible?

Comment (leoenix) [reply client/src/pages/DoctorDashboard.js]: you say the word i do the deed 
075bee470172c7717ac6189eef973bdbcb4f5f9f

**Pull request merged**